bin/plugin/commands/test-plugins.js

```diff
@@ -0,0 +1,440 @@
+		wpEnvDestinationFile: './.wp-env.override.json', // Destination .wp-env.override.json file at root level.
+		wpEnvPluginsRegexPattern: '"plugins": \\[(.*)\\],', // Regex to match plugins string in .wp-env.json.
+	} );
+};
+
+/**
+ * Handles replacement of plugins array in .wp-env.json file.
+ * Split into separate function in order to easily re-run tests with WPP plugin active.
+ *
+ * @param {WPDoReplaceWpEnvContent} settings Plugin test settings.
+ */
+function doReplaceWpEnvContent( settings ) {
+	// Regex object to match wp-env plugins string.
+	const wpEnvPluginsRegex = new RegExp( settings.wpEnvPluginsRegexPattern, 'gm' );
+
+	let wpEnvPluginsRegexReplacement = '';
+	// Amend wp-env.json to reference built plugins only.
+	// Buffer .wp-env.json content var.
+	let wpEnvFileContent = '';
+
+	try {
+		wpEnvFileContent = fs.readFileSync( settings.wpEnvFile, 'utf-8' );
+	} catch ( e ) {
+		log( formats.error( `Error reading file "${ settings.wpEnvFile }": "${ e }"` ) );
+
+		// Return with exit code 1 to trigger a failure in the test pipeline.
+		process.exit( 1 );
+	}
+
+	// If the contents of the file were incorrectly read or exception was not captured and value is blank, abort.
+	if ( '' === wpEnvFileContent ) {
+		log(
+			formats.error(
+				`File content for "${ settings.wpEnvFile }" is empty, aborting.`
+			)
+		);
+
+		// Return with exit code 1 to trigger a failure in the test pipeline.
+		process.exit( 1 );
+	}
+
+	// If we do not have a match on the wp-env enabled plugins regex, abort.
+	if ( ! wpEnvPluginsRegex.test( wpEnvFileContent ) ) {
+		log(
+			formats.error(
+				`Unable to find plugins property/key in WP Env config file: "${ settings.wpEnvFile }". Please ensure that it is present and try agagin.`
+			)
+		);
+
+		// Return with exit code 1 to trigger a failure in the test pipeline.
+		process.exit( 1 );
+	}
+
+	// Let the user know we're re-writing the .wp-env.json file.
+	log( formats.success( `Rewriting plugins property in ${ settings.wpEnvFile }` ) );
+
+	// Attempt replacement of the plugins property in .wp-env.json file to match built plugins.
+	try {
+		// Create plugins property from built plugins.
+		wpEnvPluginsRegexReplacement = `"plugins": [ "${ settings.builtPlugins
+			.map( ( item ) => {
+				if ( '.' === item ) {
+					// Do not append build dir for root plugin.
+					return item;
+				}
+
+				return `${ settings.builtPluginsDir }${ item }`;
+			} )
+			.join( '", "' ) }" ],`;
+
+		fs.writeFileSync(
+			settings.wpEnvFile,
+			wpEnvFileContent.replace(
+				wpEnvPluginsRegex,
+				wpEnvPluginsRegexReplacement
+			)
+		);
+	} catch ( e ) {
+		log(
+			formats.error(
+				`Error replacing content in ${ settings.wpEnvFile } using regex "${ wpEnvPluginsRegex }": "${ e }"`
+			)
+		);
+
+		// Return with exit code 1 to trigger a failure in the test pipeline.
+		process.exit( 1 );
+	}
+
+	// Copy the newly modified ./plugins-tests/.wp-env.json file to the root level.
```

Comment:
Why not write directly to this file rather than writing to the original and then copying?

Reply:
@joemcgill we have to copy it regardless the first time, but I agree that a check for file existing and then a copy over and final write is probably best, as if it's existing already, we may as well write to it. To be clear, it won't exist already as we need to create this overrides file, so I am not actually sure how much of a difference this makes.